drivers/spi/spi_ll_stm32.c

```diff
@@ -310,6 +310,78 @@ static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 	}
 }
 
+/* Shift a SPI frame as half-duplex master. */
+static void spi_stm32_shift_m_half(SPI_TypeDef *spi, struct spi_stm32_data* data) {
+#if defined(CONFIG_SOC_SERIES_STM32MP1X) || defined(CONFIG_SOC_SERIES_STM32H7X) ||                 \
+	defined(CONFIG_SOC_SERIES_STM32U5X)
```

Comment:
with line break: 
```
#if defined(CONFIG_SOC_SERIES_STM32MP1X) || \
        defined(CONFIG_SOC_SERIES_STM32H7X) || \
	defined(CONFIG_SOC_SERIES_STM32U5X)
```

Reply:
Please use "st,stm32h7-spi", cf https://github.com/zephyrproject-rtos/zephyr/pull/46852 which was merged recently

Reply:
Question about the formatting - is there a particular `clang-format` version that's needed, because the zephyr `.clang-format` file does not seem to agree with a lot of the formatting in this file.